flink-core/src/main/java/org/apache/flink/metrics/groups/AbstractMetricGroup.java

```diff
@@ -146,7 +147,16 @@ public Counter counter(int name) {
 
 	@Override
 	public Counter counter(String name) {
-		Counter counter = new Counter();
+		return counter(name, new SimpleCounter());
+	}
+	
+	@Override
+	public <C extends Counter> C counter(int name, C counter) {
```

Comment:
What's the advantage of introducing the bounded generic parameter over simply having a `Counter` parameter?